clients/src/main/java/org/apache/kafka/common/record/RecordBatch.java

```diff
@@ -0,0 +1,212 @@
+     * up-converting from an older message format.
+     */
+    long NO_PRODUCER_ID = -1L;
+    short NO_PRODUCER_EPOCH = -1;
+    int NO_SEQUENCE = -1;
+
+    /**
+     * Used to indicate an unknown leader epoch, which will be the case when the record set is
+     * first created by the producer.
+     */
+    int UNKNOWN_PARTITION_LEADER_EPOCH = -1;
+
+    /**
+     * Check whether the checksum of this entry is correct.
+     *
+     * @return true If so, false otherwise
+     */
+    boolean isValid();
+
+    /**
+     * Raise an exception if the checksum is not valid.
+     */
+    void ensureValid();
+
+    /**
+     * Get the checksum of this entry, which covers the entry header as well as all of the records.
+     *
+     * @return The 4-byte unsigned checksum represented as a long
+     */
+    long checksum();
+
+    /**
+     * Get the timestamp of this entry. This is the max timestamp among all records contained in this log entry.
+     *
+     * @return The max timestamp
+     */
+    long maxTimestamp();
+
+    /**
+     * Get the timestamp type of this entry. This will be {@link TimestampType#NO_TIMESTAMP_TYPE}
+     * if the message has magic 0.
+     *
+     * @return The timestamp type
+     */
+    TimestampType timestampType();
+
+    /**
+     * Get the first offset contained in this log entry. For magic version prior to 2, this generally
+     * requires deep iteration and will return the offset of the first record in the message set. For
+     * magic version 2 and above, this will return the first offset of the original message set (i.e.
+     * prior to compaction). For non-compacted topics, the behavior is equivalent.
+     *
+     * Because this requires deep iteration for older magic versions, this method should be used with
+     * caution. Generally {@link #lastOffset()} is safer since access is efficient for all magic versions.
+     *
+     * @return The base offset of this message set (which may or may not be the offset of the first record
+     *         as described above).
+     */
+    long baseOffset();
+
+    /**
+     * Get the last offset in this record set (inclusive).
+     *
+     * @return The offset of the last record in this entry
+     */
+    long lastOffset();
+
+    /**
+     * Get the offset following this entry (i.e. the last offset contained in this entry plus one).
+     *
+     * @return the next consecutive offset following this entry
+     */
+    long nextOffset();
+
+    /**
+     * Get the message format version of this entry (i.e its magic value).
+     *
+     * @return the magic byte
+     */
+    byte magic();
+
+    /**
+     * Get the producer ID (PID) for this log entry. For older magic versions, this will return 0.
+     *
+     * @return The PID or 0 if there is none
+     */
+    long producerId();
+
+    /**
+     * Get the producer epoch for this log entry.
+     *
+     * @return The producer epoch, or 0 if there is none
+     */
+    short producerEpoch();
```

Comment:
Is it right that `0` is used if there is none for this and `producerId`? The constants at the top of the file seem to indicate that it's `-1`?